src/EventSourcingDb/Types/CloudEvent.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+
+namespace EventSourcingDb.Types;
+
+internal record CloudEvent(
+    string SpecVersion,
+    string Id,
+    string Time,
+    string Source,
+    string Subject,
+    string Type,
+    string DataContentType,
+    object Data,
+    string Hash,
+    string PredecessorHash,
+    string? TraceParent = null,
+    string? TraceState = null
+)
+{
+    internal Event ToEvent() =>
+        new Event(
+            SpecVersion,
+            Id,
+            DateTimeOffset.Parse(Time),
+            Source,
+            Subject,
+            Type,
+            DataContentType,
+            Data,
+            Hash,
+            PredecessorHash,
+            TraceParent,
+            TraceState
+        );
```

Comment:
Hmmmm … kann man so machen.

Ich persönlich hätte es wahrscheinlich eher als statische `FromCloudEvent`-Funktion am `Event` definiert, oder dort einen Konstruktor definiert, dem man ein `CloudEvent` übergeben kann, aber das ist letztlich Geschmackssache.

Grundsätzlich finde ich die `From…`-Variante schöner als die `To…`-Variante, aber das ist nur meine subjektive Meinung.